linkerd/app/admin/src/server/log/stream.rs

```diff
@@ -0,0 +1,160 @@
+use futures::FutureExt;
+use hyper::{
+    body::{Buf, Bytes},
+    header, Body, StatusCode,
+};
+use linkerd_app_core::{
+    trace::{self},
+    Error,
+};
+use trace::EnvFilter;
+use tracing::instrument::WithSubscriber;
+
+static JSON_MIME: &str = "application/json";
+static JSON_HEADER_VAL: http::HeaderValue = http::HeaderValue::from_static(JSON_MIME);
+
+macro_rules! recover {
+    ($thing:expr, $msg:literal, $status:expr $(,)?) => {
+        match $thing {
+            Ok(val) => val,
+            Err(error) => {
+                tracing::warn!(%error, status = %$status, message = %$msg);
+                let json = serde_json::to_vec(&serde_json::json!({
+                    "error": error.to_string(),
+                    "status": $status.as_u16(),
+                }))?;
+                return Ok(mk_rsp($status, json));
+            }
+        }
+    }
+}
+
+// If log streaming support is enabled, start a new log stream
+pub async fn serve<B>(
+    handle: trace::Handle,
```

Comment:
Changed the type here to avoid taking a generic parameter. Cloning the handle should be cheap (and this isn't really allocation-sensitive)